pkg/app/piped/cloudprovider/kubernetes/diff_test.go

```diff
@@ -184,3 +186,145 @@ func TestDiffByCommand(t *testing.T) {
 		})
 	}
 }
+
+func TestDiff(t *testing.T) {
+	testcases := []struct {
+		name          string
+		manifests     string
+		expected      string
+		diffNum       int
+		falsePositive bool
+	}{
+		{
+			name: "Secret no diff 1",
+			manifests: `apiVersion: apps/v1
+kind: Secret
+metadata:
+  name: secret-management
+---
+apiVersion: apps/v1
+kind: Secret
+metadata:
+  name: secret-management
+`,
+			expected: "",
+			diffNum:  0,
+		},
+		{
+			name: "Secret no diff 2",
+			manifests: `apiVersion: apps/v1
+kind: Secret
+metadata:
+  name: secret-management
+data:
+  password: hoge
+stringData:
+  foo: bar
+---
+apiVersion: apps/v1
+kind: Secret
+metadata:
+  name: secret-management
+data:
+  password: hoge
+stringData:
+  foo: bar
+`,
+			expected: "",
+			diffNum:  0,
+		},
+		{
+			name: "Secret no diff with merge",
+			manifests: `apiVersion: apps/v1
+kind: Secret
+metadata:
+  name: secret-management
+data:
+  password: hoge
+stringData:
+  foo: bar
+---
+apiVersion: apps/v1
+kind: Secret
+metadata:
+  name: secret-management
+data:
+  password: hoge
+  foo: YmFy
+`,
+			expected: "",
+			diffNum:  0,
+		},
+		{
+			name: "Secret no diff override false-positive",
+			manifests: `apiVersion: apps/v1
+kind: Secret
+metadata:
+  name: secret-management
+data:
+  password: hoge
+  foo: Zm9v
+stringData:
+  foo: bar
+---
+apiVersion: apps/v1
+kind: Secret
+metadata:
+  name: secret-management
+data:
+  password: hoge
+  foo: YmFy
+`,
+			expected:      "",
+			diffNum:       0,
+			falsePositive: true,
+		},
```

Comment:
This case should not produce diffs, but it does.

Reply:
I mean this PR is not complete solution.
https://github.com/pipe-cd/pipecd/pull/3299#discussion_r814429144

Reply:
Thanks for your explanation.
I think Kubernetes does not expect the use of the same key in both `data` and `stringData` fields.
So we can simply remove this test case and don't have to care about it.
What do you think?

Reply:
[the k8s docs](https://github.com/pipe-cd/pipecd/pull/3299#discussion_r81349050) says 

> If a key appears in both the data and the stringData field, the value specified in the stringData field takes precedence.

Reply:
It's nice not to consider it for simplicity's sake.
However, I think it would be better to include them in the release notes

Reply:
I see. That is just a note about what field will be used.  As their doc, the value from `data` field will be overridden by the one in `stringData` so the value specified in the `data` field is not valid, not being used.
I mean users can specify the same key in both fields but it is meaningless so we don't have to care about that.
The diff still should be shown to be noticed.

Reply:
Thank you. 

> The diff still should be shown to be noticed.

What does this mean?